anvil/src/eth/backend/mem/fork_db.rs

```diff
@@ -118,49 +137,50 @@ impl ForkedDatabase {
 
 impl Database for ForkedDatabase {
     fn basic(&mut self, address: Address) -> AccountInfo {
-        self.backend.basic(address)
+        self.cache_db.basic(address)
```

Comment:
Should we add a test to ensure this is implemented correctly? Eg same one as the one in the issue?

Reply:
added test for scenario #1531 which ensures that modifications don't end up in the database that will be flushed to cache file